Sources/GRPCProtobufCodeGen/ProtobufCodeGenParser.swift

```diff
@@ -173,20 +173,27 @@ extension FileDescriptor {
 }
 
 extension SwiftProtobufNamer {
-  internal func trimTrailingUnderscores(from name: String) -> String {
-    var nameCopy = name
-    while nameCopy.hasSuffix("_") {
-      nameCopy = String(nameCopy.dropLast())
-    }
-    return nameCopy
-  }
-
   internal func formattedUpperCasePackage(file: FileDescriptor) -> String {
     let unformattedPackage = self.typePrefix(forFile: file)
-    return trimTrailingUnderscores(from: unformattedPackage)
+    return unformattedPackage.trimTrailingUnderscores()
   }
 
   internal func formattedLowerCasePackage(file: FileDescriptor) -> String {
-    return file.package.lowercased().replacingOccurrences(of: ".", with: "_")
+    let upperCasePackage = self.formattedUpperCasePackage(file: file)
+    let lowerCaseFirstComponent = NamingUtils.toLowerCamelCase(
+      String(upperCasePackage.split(separator: "_")[0])
+    )
+
+    return lowerCaseFirstComponent + upperCasePackage.dropFirst(lowerCaseFirstComponent.count)
```

Comment:
IIUC, we're only lower-camel-casing the first component here. I assume `toLowerCamelCase` lower-camel-cases each component in the name?

Reply:
NamingUtils.toLowerCamelCase() gets rid of the snake case, do you have a suggestion how to implement this in a nicer way (I don't really like this implementation either). Also, I've discovered that this function results in an error for empty packages, so there is still work to do

Reply:
I think I'd go for something like the following (i.e. lower camel case each component)

```swift
let upperCasePackage = self.formattedUpperCasePackage(file: file)
let lowerCaseComponents = upperCasePackage.split(separator: "_").map { component in 
  NamingUtils.toLowerCamelCase(component)
}
return lowerCaseComponents.joined(separator: "_")
```

Reply:
Thanks!